www/components/messagebox.tsx

```diff
@@ -1,73 +1,63 @@
-import { useCallback, useState } from "react";
+import { useState } from "react";
 import Image from "next/image";
 import icon from "@/public/bloomicon.jpg";
 import usericon from "@/public/usericon.svg";
 import Skeleton from "react-loading-skeleton";
-import { FaLightbulb, FaThumbsDown, FaThumbsUp } from "react-icons/fa";
+import { FaLightbulb } from "react-icons/fa";
 import { API } from "@/utils/api";
-import useSWR, { mutate } from "swr";
 
 interface MessageBoxProps {
   isUser?: boolean;
+  userId?: string;
+  URL?: string;
   messageId?: string;
   conversationId?: string;
-  api?: API;
   text: string;
   loading?: boolean;
   isThoughtsOpen?: boolean;
   setIsThoughtsOpen: (isOpen: boolean) => void;
   setThought: (thought: string) => void;
 }
 
-const fetchThought = async (key: string) => {
-  const [_, conversationId, messageId, userId] = key.split(":");
-  const api = new API({ url: process.env.NEXT_PUBLIC_API_URL!, userId });
-  return api.getThoughtById(conversationId, messageId);
-};
-
 export default function MessageBox({
   isUser,
+  userId,
+  URL,
   messageId,
-  api,
   text,
   loading = false,
   setIsThoughtsOpen,
   conversationId,
   setThought,
 }: MessageBoxProps) {
-  const [shouldFetch, setShouldFetch] = useState(false);
+  const [isThoughtLoading, setIsThoughtLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
```

Comment:
What is the error state used for here?

Reply:
Currently just displaying error text in the UI. Could hook this up to the toast notification instead? Should not be encountered in normal circumstances.